### DISCUSSION TITLE
How to run Nostr Console from your own web browser on desktop or phone

### DISCUSSION THREAD

**Discussion by vishalxl** [vishalxl/nostr_console#18] [General]
The application [nostr terminal](https://github.com/cmdruid/nostr-terminal) allows you to connect to a terminal running on a computer from any web browser. This application can be used to connect to a laptop which has a nostr_console binary, and then one can run your nostr_console from the browser. 

The actual steps are listed on its home page. 

Example results are [seen here](https://twitter.com/vishalxl/status/1590661018975637507).

For best results
1. nostr-terminal needs nodejs/node> 12.0 . This [page shows](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04) steps to install it on Linux ( or wsl on windows).
2. use the --width 72 option to give a width of 72 columns to nostr console. 
3. Create a shell script with the command line for nostr_console with all arguments, so that you can just invoke the script. Rather than having to type the full command line. 

## Running remotely using Docker

There is a file called [Dockerfile.remote](https://github.com/vishalxl/nostr_console/blob/main/Dockerfile.remote) which can be used to do this in Docker.

Use Nostr Terminal + Nostr Console to run Nostr Console remotely from a  browser. 

Build using 
```
docker build  -f Dockerfile.remote -t nostr_console_remote .
```

Then run using
```
docker run -it nostr_console_remote start
```

Do keep security/privacy factors in mind. The link is basically an online backdoor into the local terminal. Related info [here](https://github.com/vishalxl/nostr_console/discussions/18)

## Screenshots

 
![FhM42TnXkAM48Y_](https://user-images.githubusercontent.com/64505169/201167268-070156ac-0764-451a-9c8e-45216e323d1b.png)

**Comment by ghost** [comment]
Can we port this to iMessage app?
